torch/package/package_exporter.py

```diff
@@ -82,61 +93,30 @@ class PackagingError(Exception):
     """This exception is raised when there is an issue with exporting a package.
     ``PackageExporter`` will attempt to gather up all the errors and present
     them to you at once.
-
-    To make error information more understandable, the exception message will
-    only show modules that you ``intern``'d or direct dependencies of
-    ``intern``'d modules. To see the full list of error modules, consult the
-    attributes on this exception.
-
-    Attributes:
-        denied (Set[str]): modules that have been marked as denied by the exporter.
-        broken (Dict[str, str]): modules for which the exporter could not retrieve source info,
-            along with the reason that retrieving it failed.
-        unhandled (Set[str]): modules for which there is no user-specified action.
     """
 
-    def __init__(
-        self,
-        denied: Set[str],
-        broken: Dict[str, str],
-        unhandled: Set[str],
-        include_filter: Set[str],
-    ):
-        self.denied = denied
-        self.broken = broken
-        self.unhandled = unhandled
-
-        self.filtered_denied = {
-            module for module in self.denied if module in include_filter
-        }
-        self.filtered_broken = {
-            module: reason
-            for module, reason in self.broken.items()
-            if module in include_filter
-        }
-        self.filtered_unhandled = {
-            module for module in self.unhandled if module in include_filter
-        }
-
-        message = io.StringIO("Errors raised while packaging:")
-
-        if self.filtered_denied:
-            message.write(
-                "\n\n* The following modules were detected as dependencies but have been denied:\n"
-                f"{textwrap.indent(pprint.pformat(self.filtered_denied), prefix='  ')}"
-            )
-        if self.filtered_broken:
-            message.write(
-                "\n\n* The following modules did not have source information. "
-                "Extern, mock, or refactor to remove the dependency:\n"
-                f"{textwrap.indent(pprint.pformat(self.filtered_broken), prefix='  ')}"
-            )
-        if self.filtered_unhandled:
-            message.write(
-                "\n\n* The following modules did not match against any patterns. "
-                "Intern, extern, or mock them:\n"
-                f"{textwrap.indent(pprint.pformat(self.filtered_unhandled), prefix='  ')}"
-            )
+    def __init__(self, dependency_graph: DiGraph):
+        # Group errors by reason.
+        broken: Dict[PackagingErrorReason, List[str]] = {}
+        for module_name, attrs in dependency_graph.nodes.items():
+            if "error" not in attrs:
+                continue
+            error = attrs["error"]
+            if error == PackagingErrorReason.NO_ACTION:
+                assert "action" not in attrs
```

Comment:
Light reminder to make these attribute strings constants or something of that nature.